dashboard/modules/job/tests/test_job_agent.py

```diff
@@ -485,5 +486,37 @@ async def _check_all_jobs_log():
     assert all(job_check_status), job_check_status
 
 
+def test_agent_logs_not_streamed_to_drivers():
+    """Ensure when the job submission is used,
+    (ray.init is called from an agent), the agent logs are
+    not streamed to drivers.
+
+    Related: https://github.com/ray-project/ray/issues/29944
+    """
+    script = """
+import ray
+from ray.job_submission import JobSubmissionClient, JobStatus
+from ray._private.test_utils import format_web_url
+from ray._private.test_utils import wait_for_condition
+
+ray.init()
+address = ray._private.worker._global_node.webui_url
+address = format_web_url(address)
+client = JobSubmissionClient(address)
+submission_id = client.submit_job(entrypoint="ls")
+wait_for_condition(
+    lambda: client.get_job_status(submission_id) == JobStatus.SUCCEEDED
+)
+    """
+
+    proc = run_string_as_driver_nonblocking(script)
+    out_str = proc.stdout.read().decode("ascii")
+    err_str = proc.stderr.read().decode("ascii")
+
+    print(out_str, err_str)
+    assert "(raylet)" not in out_str
+    assert "(raylet)" not in err_str
```

Comment:
I've found it convenient to use the pytest `capsys` fixture for this kind of test, you don't have to use a string as the driver script so it's easier to debug failures in the script. (No need to change it in this PR)